TransformationSystem/Agent/TaskManagerAgentBase.py

```diff
@@ -209,6 +216,15 @@ def _selectTransformations( self, transType = [], status = ['Active', 'Completin
       self.log.verbose( "No transformations found" )
     else:
       self.log.verbose( "Obtained %d transformations" % len( res['Value'] ) )
+
+    if (self.moduloNumber >= 2 and self.moduloResidual >= 0) :
+      if self.moduloResidual >= self.moduloNumber :
+        self.log.error("Residual %d is equal or greater than the modulo number %d, this cannot be, check the configuration of this agent" % (self.moduloResidual, self.moduloNumber))
```

Comment:
you maybe want to do this test in the agent initialization and return an error so that the agent stops. It avoids checking every time, and then loop over empty lists of production IDs